### ISSUE TITLE
Breakout Example is stuttering

### ISSUE THREAD

**Issue by nonunknown** [bevyengine/bevy#2349]
## Bevy version

0.5

## Operating system & version

Manjaro Gnome

## What you did

Copied this example: https://github.com/bevyengine/bevy/blob/main/examples/game/breakout.rs

## What you expected to happen

Game to run smoothly and without bugs

## What actually happened

The square is jittering, this sometimes causes collision detection to bug, and it gets out of screen.

## Additional information

https://imgur.com/j52I6Kz

**Comment by alice-i-cecile** [comment]
Can you reproduce this with #2094?

**Comment by mockersf** [comment]
By "jittering", do you mean when the square is stuck between the paddle and the wall on the right? This should have been fixed by #685 but can still happen depending on your frame rate...

**Comment by nonunknown** [comment]
@alice-i-cecile I'll try

@mockersf nope, its the movement in general. Even if the FPS is low, for example 40, the square should not be jittering, because games doesnt work this way, the fixed timestep is one solution for this, but seems its not working?

**Comment by nonunknown** [comment]
@alice-i-cecile I tested here, and got the same result!

also I runned: `cargo run --release`

The game runned worse than the debug version, with way more jittery!

**Comment by cart** [comment]
I think theres a bug in #2094 that would make it stutter more. It moves the paddle without scaling it according to delta time, so framerate directly affects how fast it moves.

**Comment by alice-i-cecile** [comment]
@nonunknown I've updated #2094; is this resolved for you? I didn't notice any stuttering, but that may just be poor perception. A GIF (plus frame time information) may be useful for reproduction if this still persists.

**Comment by nonunknown** [comment]
tested it:

https://imgur.com/aQRCCez

You can compare the movement with the mouse movement, you can see how jittery it is.

**Comment by cart** [comment]
Yup thats pretty nasty. I'm also getting it to a lesser degree. I'll see what I can do. This behavior definitely isn't acceptable.

**Comment by l0b0** [comment]
Seeing the same on NixOS 21.05, rustup 1.24.2, rustc 1.54.0, Linux kernel 5.10.60, AMD Radeon RX 5500 graphics card.

**Comment by returnDanilo** [comment]
I'm experiencing the same issue. I noticed the performance increased significantly with the `WGPU_ALLOW_UNDERLYING_NONCOMPLIANT_ADAPTER=1`  env var set, so maybe it's related to #18208. (although my mesa version is *not* 25. It's 24.2.8)

Making the window size tiny also yields significant frame rate improvement.

I'll also note that in my case, I'm using Intel's Integrated graphics; no dedicated graphics card.

**Comment by Elabajaba** [comment]
> I'll also note that in my case, I'm using Intel's Integrated graphics; no dedicated graphics card.

Just wondering what CPU/GPU do you have? There's some Intel iGPUs that can run Vulkan apps but aren't [conformant](https://www.khronos.org/conformance/adopters/conformant-products/vulkan) (meaning they haven't passed the full vulkan conformance test suite, sometimes due to drivers needing more time+work eg. AMD 9070xt + RADV driver isn't conformant yet, other times due to hardware that isn't fully Vulkan capable eg. Intel Haswell/4000 series).

**Comment by returnDanilo** [comment]
> Just wondering what CPU/GPU do you have? 

The cpu is an i3-3220 which has a E3-1200 v2.  It doesn't seem to be on the list.

Running the breakout demo without extra env var flags does seem to select vulkan as the backend:

```
MESA-INTEL: warning: Ivy Bridge Vulkan support is incomplete
2025-06-24T02:17:52.083147Z  INFO bevy_render::renderer: AdapterInfo { name: "llvmpipe (LLVM 19.1.1, 256 bits)", vendor: 65541, device: 0, device_type: Cpu, driver: "llvmpipe", driver_info: "Mesa 24.2.8-1ubuntu1~24.04.1 (LLVM 19.1.1)", backend: Vulkan }
2025-06-24T02:17:52.083179Z  WARN bevy_render::renderer: The selected adapter is using a driver that only supports software rendering. This is likely to be very slow. See https://bevyengine.org/learn/errors/b0006/
```

And running it with the  `WGPU_BACKEND=gl` env var results in the app crashing:

```
2025-06-24T02:17:02.248023Z  INFO bevy_diagnostic::system_information_diagnostics_plugin::internal: SystemInfo { os: "Linux 22.1 Linux Mint", kernel: "6.8.0-62-generic", cpu: "Intel(R) Core(TM) i3-3220 CPU @ 3.30GHz", core_count: "2", memory: "19.4 GiB" }

thread 'main' panicked at crates/bevy_render/src/renderer/mod.rs:196:10:
Unable to find a GPU! Make sure you have installed required drivers! For extra information, see: https://github.com/bevyengine/bevy/blob/latest/docs/linux_dependencies.md
```
